<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsNewGeoPackageLayerDialogBase</class>
 <widget class="QDialog" name="QgsNewGeoPackageLayerDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>548</width>
    <height>686</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>351</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>New GeoPackage Layer</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="9" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Help|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>530</width>
        <height>635</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="4" column="0">
        <widget class="QCheckBox" name="mCheckBoxCreateSpatialIndex">
         <property name="toolTip">
          <string>Add an integer id field as the primary key for the new layer</string>
         </property>
         <property name="text">
          <string>Create a spatial index</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_6"/>
       </item>
       <item row="5" column="0">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>New field</string>
         </property>
         <layout class="QGridLayout" name="_2">
          <item row="0" column="1" colspan="3">
           <widget class="QLineEdit" name="mFieldNameEdit"/>
          </item>
          <item row="1" column="1" colspan="3">
           <widget class="QComboBox" name="mFieldTypeBox"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="mFieldLengthLabel">
            <property name="text">
             <string>Maximum length</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="mFieldNameLabel">
            <property name="text">
             <string>Name</string>
            </property>
            <property name="buddy">
             <cstring>mFieldNameEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <widget class="QToolButton" name="mAddAttributeButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Add field to list</string>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Add to fields list</string>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>../../images/themes/default/mActionNewAttribute.svg</normaloff>../../images/themes/default/mActionNewAttribute.svg</iconset>
            </property>
            <property name="toolButtonStyle">
             <enum>Qt::ToolButtonTextBesideIcon</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="mFieldTypeLabel">
            <property name="text">
             <string>Type</string>
            </property>
            <property name="buddy">
             <cstring>mFieldTypeBox</cstring>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="3">
           <widget class="QLineEdit" name="mFieldLengthEdit">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Field length / width&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="4" column="2">
          <widget class="QLineEdit" name="mLayerIdentifierEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Human-readable identifier (e.g. short name) for the layer content&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="8" column="2">
          <widget class="QLineEdit" name="mGeometryColumnEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Name of the geometry column&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>geometry</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLineEdit" name="mDatabaseEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Existing or new GeoPackage database file name&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="mSelectDatabaseButton">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select an existing or create a new database&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="mLayerDescriptionLabel">
           <property name="text">
            <string>Layer description</string>
           </property>
           <property name="buddy">
            <cstring>mLayerIdentifierEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="mGeometryTypeLabel">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Geometry type</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="mGeometryColumnLabel">
           <property name="text">
            <string>Geometry column</string>
           </property>
           <property name="buddy">
            <cstring>mGeometryColumnEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLineEdit" name="mTableNameEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Table name in the database&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="5" column="2">
          <widget class="QLineEdit" name="mLayerDescriptionEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Human-readable description for the layer content&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="mDatabaseLabel">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Database</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="mTableNameLabel">
           <property name="text">
            <string>Table name</string>
           </property>
           <property name="buddy">
            <cstring>mLayerIdentifierEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="mLayerIdentifierLabel">
           <property name="text">
            <string>Layer identifier</string>
           </property>
           <property name="buddy">
            <cstring>mLayerIdentifierEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="7" column="2">
          <widget class="QComboBox" name="mGeometryTypeBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Geometry type&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="mFeatureIdLabel">
           <property name="text">
            <string>Feature id column</string>
           </property>
           <property name="buddy">
            <cstring>mGeometryColumnEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="6" column="2">
          <widget class="QLineEdit" name="mFeatureIdColumnEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Name of the feature id column&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>fid</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="6" column="0">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Fields list</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="2" column="0">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QToolButton" name="mRemoveAttributeButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Delete selected field</string>
            </property>
            <property name="text">
             <string>Remove field</string>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>../../images/themes/default/mActionDeleteAttribute.svg</normaloff>../../images/themes/default/mActionDeleteAttribute.svg</iconset>
            </property>
            <property name="toolButtonStyle">
             <enum>Qt::ToolButtonTextBesideIcon</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <widget class="QTreeWidget" name="mAttributeView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="rootIsDecorated">
             <bool>false</bool>
            </property>
            <property name="columnCount">
             <number>3</number>
            </property>
            <column>
             <property name="text">
              <string>Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Type</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Length</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QgsProjectionSelectionWidget" name="mCrsSelector" native="true">
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QgsProjectionSelectionWidget</class>
   <extends>QWidget</extends>
   <header location="global">qgsprojectionselectionwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mDatabaseEdit</tabstop>
  <tabstop>mSelectDatabaseButton</tabstop>
  <tabstop>mTableNameEdit</tabstop>
  <tabstop>mLayerIdentifierEdit</tabstop>
  <tabstop>mLayerDescriptionEdit</tabstop>
  <tabstop>mFeatureIdColumnEdit</tabstop>
  <tabstop>mGeometryTypeBox</tabstop>
  <tabstop>mGeometryColumnEdit</tabstop>
  <tabstop>mCrsSelector</tabstop>
  <tabstop>mCheckBoxCreateSpatialIndex</tabstop>
  <tabstop>mFieldNameEdit</tabstop>
  <tabstop>mFieldTypeBox</tabstop>
  <tabstop>mFieldLengthEdit</tabstop>
  <tabstop>mAddAttributeButton</tabstop>
  <tabstop>mAttributeView</tabstop>
  <tabstop>mRemoveAttributeButton</tabstop>
  <tabstop>buttonBox</tabstop>
  <tabstop>scrollArea</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsNewGeoPackageLayerDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>290</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
